Sources/DatadogSDKTesting/DDTestMonitor.swift

```diff
@@ -128,4 +156,21 @@ internal class DDTestMonitor {
     func stopStderrCapture() {
         stderrCapturer.stopCapturing()
     }
+
+    func startAttributeListener() {
+        func attributeCallback(port: CFMessagePort?, msgid: Int32, data: CFData?, info: UnsafeMutableRawPointer?) -> Unmanaged<CFData>? {
+            if let data = data as Data? {
+                let decoded = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any]
+                decoded?.forEach {
+                    DDTestMonitor.instance?.currentTest?.setTag(key: $0.key, value: $0.value)
+                }
+            }
+
+            return nil
+        }
+
+        let port = CFMessagePortCreateLocal(nil, "DatadogTestingPort" as CFString, attributeCallback, nil, nil)
+        let runLoopSource = CFMessagePortCreateRunLoopSource(nil, port, 0)
+        CFRunLoopAddSource(CFRunLoopGetCurrent(), runLoopSource, CFRunLoopMode.commonModes)
```

Comment:
Ok, here's the receiver. Is it that simple to talk through Mach ports or some complexity is not visible in this PR?

Reply:
Yes, it is that simple. As a general solution for IPC ( not on UI tests) sandboxing will make it fail for sure.